river/stream/twitch_chat_stream.py

```diff
@@ -0,0 +1,167 @@
+from datetime import datetime
+from typing import Optional, Iterator, List
+from enum import Enum, auto
+from dataclasses import dataclass, asdict
+import socket
+import re
+
+SERVER = "irc.chat.twitch.tv"
+PORT = 6667
+BUFFER_SIZE = 2048  # bytes
+ENCODING = "utf-8"
+TIMEOUT = 5  # seconds
+CHAT_ITEM_PATTERN = r":(.*)\!.*@.*\.tmi\.twitch\.tv PRIVMSG #(.*) :(.*)"
+PING_PATTERN = r"\bPING :tmi\.twitch\.tv\b"
+DT_FORMAT = "%Y-%m-%dT%H:%M:%S"
+
+
+class IrcMessage(Enum):
+    PASS = auto()
+    NICK = auto()
+    JOIN = auto()
+    PONG = auto()
+
+
+@dataclass
+class ChatMessageItem:
+    dt: datetime
+    channel: str
+    username: str
+    msg: str
+
+
+class TwitchChatStream:
+    """Twitch chat stream client.
+
+    This client gives access to a live stream of chat messages in Twitch channels using IRC protocol.
+    You need to have a Twitch account and receive an OAuth token from https://twitchapps.com/tmi/.
+
+    Parameters
+    ----------
+        nickname
+            The nickname of your account.
+        token
+            OAuth token which has been generated.
+        channels
+            A list of channel names like ["asmongold", "shroud"] you want to collect messages from.
+        buffer_size
+            Size of buffer in kB used for receiving responses from Twitch with IRC (default 2 kB).
+        timeout
+            A timeout value in seconds for waiting response from Twitch (default 5s). It can be usefull if all channels requested are offline.
```

Comment:
```suggestion
            A timeout value in seconds for waiting response from Twitch (default 5s). It can be useful if all requested channels are offline.
```